### PR TITLE
Adopting boolean module

### PR DESCRIPTION
**Pull request by dorafmon**
Could you please review? @bordoley @glennsl  Thanks!

### CODE REVIEW

**Comment by bordoley** [comment]
Not directly related, but why are you prefixing the file/module names with b. I only did this in Imm.re for modules that conflicted with existing stdlib names. Also, personally I've decided not to define mli files per private modules and instead expose a single top level module in as single mli. This allowed me to consolidate all documentation in a single place, and also had the nice side effect of allowing my public api to have recursive dependencies which are not actually possible in the underlying definition, because the underlying functions are defined in different modules. For instance, using the private List module in Iterable, while the private List module has a public toIterable function.

File: src/types/bbool.ml [hunk 1]
```diff
@@ -0,0 +1,17 @@
+
+(* This file is free software, part of containers. See file "license" for more details. *)
+
+type t = bool
+
+let negate x = not x;;
+
+let equals = fun a b -> match a, b with
```
Comment (bordoley) [inline src/types/bbool.ml]: Why not?
```
let equals = fun a b -> a == b
```

File: src/types/bbool.mli [hunk 2]
```diff
@@ -5,14 +5,8 @@
...
-
-val equal : t -> t -> bool
-
 val negate : t -> t
 (** Negation on booleans (functional version of [not]) *)
 
-type 'a printer = Format.formatter -> 'a -> unit
-
-val pp : t printer
+include Equatable.S with type t := t
```
Comment (bordoley) [inline src/types/bbool.mli]: Comparable infer Equatable.

**Comment by dorafmon** [comment]
@bordoley Thanks for the review. I updated the PR according to it.

**Comment by dorafmon** [comment]
The thing is, without a .mli file, I personally find two disadvantages:
1. access control is limited, we could not have private/helper functions?
2. I cannot express that `Bool` is an instance of `Comparable`? With a .mli file I can just say `include Comparable.S with t := t`.

This is why I prefer a .mli file.

**Comment by bordoley** [comment]
@dorafmon I'm not arguing against mli files altogether. Instead, I'm arguing that you should add src/containers.mli, and in that file define all the type/module signatures. This will provide you the benefits of ensuring that your underlying implementations implement all the expected public types/signatures (Comparable, etc.) and also centralizes your documentation in a single file. It also allows you to disconnect your public interface from private implementation, such that you can define functions on modules that otherwise would not be possible otherwise to circular dependencies.

**Comment by dorafmon** [comment]
@bordoley we have a top level .mli file, it is `src/containers.mli`

**Comment by bordoley** [comment]
@dorafmon I only see a top level src/containers.ml, not mli. Is that file checked in?

**Comment by dorafmon** [comment]
@bordoley ah yes that is the file I am talking about, I agree with you that it is beneficial to have a top level module (in our case it is containers.ml, I will provide a core.ml shortly afterward to isolate out the core modules).

**Pull request merged**